app/src/test/java/com/example/countryinfo/CountryDetailsViewModelTest.kt

```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Test
+    fun `when countryClicked invoked and countryDetails is null then post error CountryDetailsViewState on success`() {
+        val mockResponse: Response<GetCountryByIdQuery.Data> =
+            mock(Response::class.java) as Response<GetCountryByIdQuery.Data>
+        val mockData = mock(GetCountryByIdQuery.Data::class.java, Answers.RETURNS_DEEP_STUBS)
+
+        `when`(mockResponse.data).thenReturn(mockData)
+        `when`(mockData.country!![0]!!.fragments.countryDetails).thenReturn(null)
+        `when`(countriesApiMock.getCountryById(testCountryId)).thenReturn(
+            Observable.just(
+                mockResponse
+            )
+        )
+
+        val mockObserver = mock(Observer::class.java) as Observer<CountryDetailsViewState>
+        viewModel.countryLiveData.observeForever(mockObserver)
+
+        viewModel.countryClicked(testCountryId)
+
+        val argumentCaptor = ArgumentCaptor.forClass(CountryDetailsViewState::class.java)
+        verify(mockObserver, times(1)).onChanged(argumentCaptor.capture())
+
+        assert(argumentCaptor.allValues.last() is CountryDetailsViewState.Error)
+
+        val actualState = argumentCaptor.allValues.last() as CountryDetailsViewState.Error
+
+        assertThat(actualState.error.message, `is`("Country is not available :("))
+
+    }
+
+    @Test
+    fun `when countryClicked invoked and response has errors then post error CountryDetailsViewState on success`() {
+        `when`(countriesApiMock.getCountryById(testCountryId)).thenReturn(
+            Observable.error(ApolloException("Test error"))
+        )
+
+        val mockObserver = mock(Observer::class.java) as Observer<CountryDetailsViewState>
+        viewModel.countryLiveData.observeForever(mockObserver)
+
+        viewModel.countryClicked(testCountryId)
+
+        val argumentCaptor = ArgumentCaptor.forClass(CountryDetailsViewState::class.java)
+        verify(mockObserver, times(1)).onChanged(argumentCaptor.capture())
+
+        assert(argumentCaptor.allValues.last() is CountryDetailsViewState.Error)
+
+        val actualState = argumentCaptor.allValues.last() as CountryDetailsViewState.Error
+
+        assertThat(actualState.error.message, `is`("Test error"))
+
+    }
+
+    private fun mockCountryDetailsFragment(): CountryDetails {
+        val mockCountryDetailsFragment =
+            mock(CountryDetails::class.java, Answers.RETURNS_DEEP_STUBS)
+        mockCountryDetailsFragment.apply {
+            `when`(name).thenReturn("Test name")
+            `when`(capital).thenReturn("Test capital")
+            `when`(subregion?.region?.name).thenReturn("Test subregion name")
+            `when`(population).thenReturn(1_000_000.0)
+            `when`(flag?.svgFile).thenReturn("https://test.com")
+            `when`(currencies).thenReturn(listOf())
+            `when`(officialLanguages).thenReturn(listOf())
+            `when`(timezones).thenReturn(listOf())
+            `when`(callingCodes).thenReturn(listOf())
```

Comment:
Let's test with non-empty lists for those values.

Reply:
Done.